<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MLPTrainingDialog</class>
 <widget class="QDialog" name="MLPTrainingDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>982</width>
    <height>593</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Entrenamiento</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinimumSize</enum>
   </property>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>962</width>
        <height>543</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Propiedades</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QFormLayout" name="formLayout">
              <property name="fieldGrowthPolicy">
               <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
              </property>
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Entradas</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="lblInputs">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::Box</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Sunken</enum>
                </property>
                <property name="text">
                 <string>TextLabel</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_2">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Salidas</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLabel" name="lblOutputs">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::Box</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Sunken</enum>
                </property>
                <property name="text">
                 <string>TextLabel</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_3">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Capas</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <property name="spacing">
                 <number>6</number>
                </property>
                <item>
                 <widget class="QTableWidget" name="tblLayers">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>150</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>16777215</width>
                    <height>16777215</height>
                   </size>
                  </property>
                  <property name="editTriggers">
                   <set>QAbstractItemView::DoubleClicked|QAbstractItemView::SelectedClicked</set>
                  </property>
                  <property name="alternatingRowColors">
                   <bool>true</bool>
                  </property>
                  <property name="rowCount">
                   <number>0</number>
                  </property>
                  <property name="columnCount">
                   <number>2</number>
                  </property>
                  <attribute name="verticalHeaderVisible">
                   <bool>false</bool>
                  </attribute>
                  <column/>
                  <column/>
                 </widget>
                </item>
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_5">
                  <item>
                   <widget class="QToolButton" name="btnAddLayer">
                    <property name="text">
                     <string>...</string>
                    </property>
                    <property name="icon">
                     <iconset resource="../imagenes.qrc">
                      <normaloff>:/imagenes/plus_icon.png</normaloff>:/imagenes/plus_icon.png</iconset>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QToolButton" name="btnDeleteLayer">
                    <property name="text">
                     <string>...</string>
                    </property>
                    <property name="icon">
                     <iconset resource="../imagenes.qrc">
                      <normaloff>:/imagenes/minus_icon.png</normaloff>:/imagenes/minus_icon.png</iconset>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <spacer name="verticalSpacer">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </item>
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="label_6">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Algoritmo principal</string>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QComboBox" name="cbTrainingAlgorithm">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <item>
                 <property name="text">
                  <string>Backpropagation</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="7" column="0">
               <widget class="QLabel" name="label_7">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Funcion de transferencia</string>
                </property>
               </widget>
              </item>
              <item row="7" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QComboBox" name="cbTrasnferFunction">
                  <item>
                   <property name="text">
                    <string>Sigmoide</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Tangente</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="8" column="0">
               <widget class="QLabel" name="label_9">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Conjunto de entrenamiento</string>
                </property>
               </widget>
              </item>
              <item row="8" column="1">
               <widget class="QPushButton" name="btnEditTrainingSet">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Editar</string>
                </property>
               </widget>
              </item>
              <item row="9" column="0">
               <widget class="QLabel" name="label_30">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Conjunto de validación</string>
                </property>
               </widget>
              </item>
              <item row="9" column="1">
               <widget class="QPushButton" name="btnEditValidationTest">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Editar</string>
                </property>
               </widget>
              </item>
              <item row="10" column="0">
               <widget class="QLabel" name="label_33">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Conjunto de prueba</string>
                </property>
               </widget>
              </item>
              <item row="10" column="1">
               <widget class="QPushButton" name="btnEditTestSet">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Editar</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <item>
            <widget class="QGroupBox" name="groupBox_2">
             <property name="minimumSize">
              <size>
               <width>300</width>
               <height>0</height>
              </size>
             </property>
             <property name="title">
              <string>Condición de parada</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="sizeConstraint">
               <enum>QLayout::SetMinimumSize</enum>
              </property>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QComboBox" name="cbStopCondition">
                  <item>
                   <property name="text">
                    <string>MSE minimo</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>RMSE minimo</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>CE minimo</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="sbEMValue">
                  <property name="decimals">
                   <number>6</number>
                  </property>
                  <property name="maximum">
                   <double>999.990000000000009</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                  <property name="value">
                   <double>5.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QPushButton" name="btnRandomize">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Aleatorizar pesos</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>Optimización</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QScrollArea" name="saOptimization">
                <property name="widgetResizable">
                 <bool>true</bool>
                </property>
                <widget class="QWidget" name="scrollAreaWidgetContents">
                 <property name="geometry">
                  <rect>
                   <x>0</x>
                   <y>0</y>
                   <width>367</width>
                   <height>313</height>
                  </rect>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_9">
                  <item>
                   <widget class="QCheckBox" name="chkSA">
                    <property name="text">
                     <string>Simulated Annealing</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="gbPlotter">
         <property name="title">
          <string>Resultados</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Tiempo</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="lblTime">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonsLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnMultipleTraining">
       <property name="text">
        <string>Multiples entrenamientos</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnTrain">
       <property name="text">
        <string>Entrenar</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="text">
        <string>Salir</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../imagenes.qrc"/>
 </resources>
 <connections/>
</ui>
